src/test/embeddingsnode_test.cpp

```diff
@@ -590,3 +590,175 @@ TEST_F(EmbeddingsInvalidTokenizerConfigTest, simpleNegative) {
+    rapidjson::ParseResult ok = d.Parse(response.c_str());
+    ASSERT_EQ(ok.Code(), 0);
+    ASSERT_TRUE(d.HasMember("tokens"));
+    ASSERT_TRUE(d["tokens"].IsArray());
+    ASSERT_EQ(d["tokens"].Size(), 100);
+}
+
+TEST_F(EmbeddingsTokenizeHttpTest, tokenizePositivePaddingSideLeft) {
+    std::string requestBody = R"(
+        {
+            "model": "embeddings_ov",
+            "text": "hello world",
+            "max_length": 100,
+            "pad_to_max_length": true,
+            "padding_side": "left"
+        }
+    )";
+    ASSERT_EQ(
+        handler->dispatchToProcessor(endpointTokenize, requestBody, &response, comp, responseComponents, writer, multiPartParser),
+        ovms::StatusCode::OK);
+    rapidjson::Document d;
+    rapidjson::ParseResult ok = d.Parse(response.c_str());
+    ASSERT_EQ(ok.Code(), 0);
+    ASSERT_TRUE(d.HasMember("tokens"));
+    ASSERT_TRUE(d["tokens"].IsArray());
+    ASSERT_EQ(d["tokens"].Size(), 100);
+    ASSERT_EQ(d["tokens"][0].GetInt(), 0);
+}
+
+TEST_F(EmbeddingsTokenizeHttpTest, tokenizePositivePaddingSideRight) {
+    std::string requestBody = R"(
+        {
+            "model": "embeddings_ov",
+            "text": "hello world",
+            "max_length": 100,
+            "pad_to_max_length": true,
+            "padding_side": "right"
+        }
+    )";
+    ASSERT_EQ(
+        handler->dispatchToProcessor(endpointTokenize, requestBody, &response, comp, responseComponents, writer, multiPartParser),
+        ovms::StatusCode::OK);
+    rapidjson::Document d;
+    rapidjson::ParseResult ok = d.Parse(response.c_str());
+    ASSERT_EQ(ok.Code(), 0);
+    ASSERT_TRUE(d.HasMember("tokens"));
+    ASSERT_TRUE(d["tokens"].IsArray());
+    ASSERT_EQ(d["tokens"].Size(), 100);
+    ASSERT_EQ(d["tokens"][99].GetInt(), 0);
+}
+
+TEST_F(EmbeddingsTokenizeHttpTest, tokenizeNegativeInvalidPaddingSide) {
+    std::string requestBody = R"(
+        {
+            "model": "embeddings_ov",
+            "text": "hello world",
+            "padding_side": "invalid_value"
+        }
+    )";
+    Status status = handler->dispatchToProcessor(endpointTokenize, requestBody, &response, comp, responseComponents, writer, multiPartParser);
+    ASSERT_EQ(status, ovms::StatusCode::MEDIAPIPE_EXECUTION_ERROR) << status.string();
+}
+
+TEST_F(EmbeddingsTokenizeHttpTest, tokenizePositiveAddSpecialTokensFalse) {
+    std::string requestBody = R"(
+        {
+            "model": "embeddings_ov",
+            "text": "hello world",
+            "add_special_tokens": false
+        }
+    )";
+    ASSERT_EQ(
+        handler->dispatchToProcessor(endpointTokenize, requestBody, &response, comp, responseComponents, writer, multiPartParser),
+        ovms::StatusCode::OK);
+    rapidjson::Document d;
+    rapidjson::ParseResult ok = d.Parse(response.c_str());
+    ASSERT_EQ(ok.Code(), 0);
+    ASSERT_TRUE(d.HasMember("tokens"));
+    ASSERT_TRUE(d["tokens"].IsArray());
+    ASSERT_EQ(d["tokens"].Size(), 2);
```

Comment:
would it be worth asserting which tokens were generated? (and in all other tests wherever reasonable length)